src/ripple/protocol/STAmount.h

```diff
@@ -104,6 +104,32 @@ class STAmount final
 
     STAmount (Issue const& issue, int mantissa, int exponent = 0);
 
+    std::size_t
+    size_of() const override
+    {
+        return sizeof(*this);
+    }
+
+    STBase*
+    copy (std::size_t n, void* buf) const override
```

Comment:
@HowardHinnant Sure, but you still need to define `copy` and `move` for every subclass. Is there a way to avoid that, perhaps using CRTP (I showed a snippet earlier today, `STBaseWrap`, but it fails when the hierarchy of inheritance is more than 2 classes deep).